### ISSUE TITLE
Sayma in uTCA: no power

### ISSUE THREAD

**Issue by jbqubit** [sinara-hw/sinara#475]
@sbourdeauducq and I both see a problem with Sayma in uTCA rack. Take a board that works fine on the desktop powered via external supply and put it into uTCA rack. JTAG over USB stops working. @gkasprow do you see this?

**Comment by gkasprow** [comment]
@hartytp is the blue LED blinking on the front panel? Did you enable the power supply in the crate?

**Comment by hartytp** [comment]
I think you mean @jbqubit

**Comment by jbqubit** [comment]
@gkasprow asked
> is the blue LED blinking on the front panel

The power for the AMC slot is enabled and the MCH power led is blinking green. When I run Sayma_AMC and Sayma_RTM on the desktop in my lab using stand-alone power supply I observe 3.3 A current. In uTCA crate the current is only 1.85 A ([as reported by MCH](https://github.com/m-labs/sinara/issues/371#issuecomment-358113664)).

**Comment by jbqubit** [comment]
And the blue LED is not blinking.

**Comment by gkasprow** [comment]
@jbqubit can you check if LEDs on Sayma AMC, close to the DC/DC converter are ON?
The same with RTM LEDs.
It looks like the RTM is not enabled. Maybe it is not fully plugged?

**Comment by gkasprow** [comment]
anyway, not initialised Sayma should consume roughly 1.9..2A

**Comment by jbqubit** [comment]
- Sayma AMC power LEDs are not on. 
- Sayma RTM power LEDs are not on.

**Comment by gkasprow** [comment]
so if the power LEDs are not on, so who consumes that much power...I will check it with my crate.

**Comment by jbqubit** [comment]
Were you able to reproduce @gkasprow?

**Comment by jbqubit** [comment]
@gkasprow Were you able to reproduce this?

**Comment by jbqubit** [comment]
Changed name to "Sayma in uTCA: no power" as this better reflects the underlying problem.

**Comment by jbqubit** [comment]
I've shorted pins 13 and 11 of JTAG header as in https://github.com/m-labs/sinara/issues/463. But no improvement.

**Comment by sbourdeauducq** [comment]
@gkasprow Have you reproduced/looked into this issue? Ken Brown is asking when the boards will work in µTCA.

**Comment by gkasprow** [comment]
We finally managed to collect all required HW in the lab. NAT MCH and the supply were lent to other group. Jakub is working on it right now trying to make OpenMMC running on Sayma. So expect progress very soon.
![2018-01-24 11 51 17](https://user-images.githubusercontent.com/4325054/35329048-ae577c08-00fe-11e8-8bb6-a194385f98fb.jpg)

**Comment by sbourdeauducq** [comment]
Porting OpenMMC in a short time sounds a bit ambitious. Maybe there is a quick hack that can be done to get the power supply running?

**Comment by gkasprow** [comment]
We port only necessary things, power supply and sensor. All remaining stuff is not really useful for ARTIQ.
The original processor for which it was developed differs mainly by package.

**Comment by gkasprow** [comment]
Jakub ported the MMC to new CPU, tomorrow we will try to make it running.

**Comment by gkasprow** [comment]
MMC works, we still need to resolve some minor issues, but the MMC gets power. Hot swap also works. Tested with NAT and Vadatech MCH.

**Comment by jbqubit** [comment]
Does Sayma AMC + RTM also power up? Can you see startup ARTIQ/MiSoC UART messages on AMC?

**Comment by gkasprow** [comment]
Didn't check yet with RTM. The AMC wakes up, all supplies are on, start-up messages are on the USB UART.

**Comment by gkasprow** [comment]
RTM wakes up as well.

**Comment by jbqubit** [comment]
Congrats on getting the OpenOCD up and running on Sayma AMC! That's a big step toward the OpenMMC [milestone](https://github.com/m-labs/sinara/milestone/27). 

As of this Issue now there's two versions of MMC software. One contains many months of configuration tweaks that got Sayma working (Forth). The second version now only contains these uTCA-power configuration tweaks (OpenMMC). Can you back port the uTCA-power tweaks to the Forth code so we can commence with testing with rack? I agree with @sbourdeauducq that it's better to reduce dependency on OpenOCD port for purposes of [Sayma v1 milestone](https://github.com/m-labs/sinara/milestone/26).

**Comment by jbqubit** [comment]
Once added to Forth MMC firmware and confirmed to work by @sbourdeauducq let's close this Issue.

**Comment by sbourdeauducq** [comment]
OpenOCD? Forth? This has close to nothing to do with the MMC.

**Comment by gkasprow** [comment]
Forth was used to play with FPG peripherals.
Jakub already integrated my init procedures with OpenMMC, now he is testing them

**Comment by jbqubit** [comment]
> OpenOCD? Forth?
Typo. I mean OpenOCD. 

OK. Please use OpenMMC [milestone](https://github.com/m-labs/sinara/milestone/27) and related Issues to track progress of Jakub's work.

**Comment by dhslichter** [comment]
>OpenOCD? Forth?
Typo. I mean OpenOCD.

You mean OpenMMC? :)

**Comment by jbqubit** [comment]
OK. To test I need to know how to program OpenMMC firmware. cf https://github.com/m-labs/sinara/issues/428

**Comment by gkasprow** [comment]
@jbqubit are you able to get the FlashMagic software running on your local machine?

**Comment by jbqubit** [comment]
Yes, I have FlashMagic on a Windows machine.

**Comment by gkasprow** [comment]
OK.
Short pin 2 of T23 with pin 2 of T24. This will enable reset over USB
Please connect USB to the computer.
In device manager/ Ports (COM&LPT) check which USB port is FTDI port D
Run FlashMagic tool, select correct COM port.
I will place *hex file here so you could test if it works.

**Comment by gkasprow** [comment]
set LPC1776, 8MHz oscillator, select hex file and press start.

**Comment by jbqubit** [comment]
Which .hex file?

**Comment by gkasprow** [comment]
I'm trying to generate it. I have it configured on another computer. I need to get hex file from axf file.

**Comment by gkasprow** [comment]
OK, this should work:
to convert axf to FlashMagic compatible format I used this command. I post it for further reference.
```
arm-none-eabi-objcopy -O binary "${BuildArtifactFileName}" "${BuildArtifactFileBaseName}.bin"
checksum -v "${BuildArtifactFileBaseName}.bin"
arm-none-eabi-objcopy -I binary "${BuildArtifactFileBaseName}.bin" -O ihex "${BuildArtifactFileBaseName}.hex"
```
Please use this file

**Comment by sbourdeauducq** [comment]
You need only one command. https://github.com/m-labs/sinara/issues/448#issuecomment-359141733

**Comment by jbqubit** [comment]
I received replacement AMC from @gkasprow this week. Board powers on fine on bench top when connected to RTM. The Blue front panel LED is not blinking. @gkasprow says this is proper behavior of board in uTCA crate -- MMC firmware on this AMC is now more standards compliant. OK good. 

@gkasprow @wizath But...... For this AMC + RTM in uTCA rack... I see fast flashing of blue, red, green lights on front panel. No power LEDS are illuminated on RTM or AMC. ​

**Comment by jbqubit** [comment]
@gkasprow @wizath On your board can you ping Sayma using MCH as switch?

**Comment by gkasprow** [comment]
@jbqubit we have troubles getting ARTIQ running on the only AMC board we have. On Tuesday I will receive your board so we will have more chances to debug it.
To diagnose Ethernet the list of the things to check are:
- ping
- LINK UP LED
- GMIICR register content in the PHY dump (E character)
If the GMIICR register is zero or all ones, it means that it was not initialised properly and there is no chance for Ethernet to work.
@wizath has new MMC with some issues solved so try it.

If the power doesn't work in the crate, just check what MMC and MCH says

**Comment by sbourdeauducq** [comment]
I flashed your openMMC_exar.axf and updated the Exar chips. The Sayma appears to work fine outside the µTCA crate.
But now the whole µTCA crate dies (no more MCH response to TCP/IP) when a Sayma is inserted and the hotswap switch closed, with the following final MCH messages:
```
ScanPM: ...
no PM for fru_id=51
AMC2(6): Handle=0x01 - closed
no PM for fru_id=52
LSHM(0): FRU 6 sensor 38 LUN 0 'HS 006 AMC2' hotswap M1->M2
LSHM(0): FRU 6 sensor 38 LUN 0 'HS 006 AMC2' hotswap M1->M2
Activation: modules are ready
Activation: all modules ready, Allowance Period (20 sec) stopped - continue with module startup !
Activation: starting AMC2
bp_fru: power chan=6, current limit=8.0 A
PwrAllocate(fru 6 chan 6): allocating 8.0 A granted (available 44.5 A)
pm_EnablePayloadPwr(50,6):
PM(50,6):Enable PP (primSite:1 secSite:0)
pm_PP_good(6): Failure: power channel state=0x0b
pm_PP_good(6): Failure: power channel state=0x0b
```

**Comment by sbourdeauducq** [comment]
And of course another board produces different behavior:
```
ipmiMsgSender(6): RSP(I2C=0x74) failed on bus 2 result -8
ipmiMsgSender(6): RSP(I2C=0x74) failed on bus 2 result -8
ipmiMsgSender(6): RSP(I2C=0x74) failed on bus 2 result -8
ipmiMsgSender(6): REQ(I2C=0x74) failed on bus 2 - no ACK
R(6,2,2)pm_processEvent PM(50): channel(1) new state=0x5b
ipmiMsgSender(6): RSP(I2C=0x74) failed on bus 2 result -8
ipmiMsgSender(6): RSP(I2C=0x74) failed on bus 2 result -8
ipmiMsgSender(6): RSP(I2C=0x74) failed on bus 2 result -8
ipmiMsgSender(6): RSP(I2C=0x74) failed on bus 2 result -8
ipmiMsgSender(6): RSP(I2C=0x74) failed on bus 2 result -8
ipmiMsgSender(6): RSP(I2C=0x74) failed on bus 2 result -8
ipmiMsgSender(6): RSP(I2C=0x74) failed on bus 2 result -8
ipmiMsgSender(6): RSP(I2C=0x74) failed on bus 2 result -8
ipmiMsgSender(6): RSP(I2C=0x74) failed on bus 2 result -8
ipmiMsgSender(6): RSP(I2C=0x74) failed on bus 2 result -8
ipmiMsgSender(6): RSP(I2C=0x74) failed on bus 2 result -8
ipmiMsgSender(6): RSP(I2C=0x74) failed on bus 2 result -8
ipmiMsgSender(6): RSP(I2C=0x74) failed on bus 2 result -8
ipmiMsgSender(6): RSP(I2C=0x74) failed on bus 2 result -8
ipmiMsgSender(6): RSP(I2C=0x74) failed on bus 2 result -8
```
...and no MCH dying.

**Comment by sbourdeauducq** [comment]
Sometimes I get yet another error:
```
Activation: modules are ready
Activation: all modules ready, Allowance Period (20 sec) stopped - continue with module startup !
Activation: starting AMC3
bp_fru: power chan=7, current limit=8.0 A
PwrAllocate(fru 7 chan 7): allocating 8.0 A granted (available 45.5 A)
pm_EnablePayloadPwr(50,7):
.CU1(40): FRU active (state M4)
PM(50,7):Enable PP (primSite:1 secSite:0)
pm_PP_good(7): Failure: power channel state=0x0b
pm_PP_good(7): Failure: power channel state=0x0b
CU1(40)  fan speed properties:
  minimum speed level:    0x00
  maximum speed level:    0x0f
  normal operating level: 0x03
  fan tray properties:    0x80
PM(50,7): new state=0x1b
pm_processEvent PM(50): channel(3) new state=0x1b
pm_processEvent PM(50): channel(7) new state=0x1b
...LSHM(0): FRU 7 sensor 38 LUN 0 'HS 007 AMC3' hotswap M2->M3
LSHM(0): CU1 FRU 40 added
LSHM(0): FRU 7 sensor 38 LUN 0 'HS 007 AMC3' hotswap M2->M3
LSHM(0): FRU 40 sensor 55 LUN 0 'Hot Swap' hotswap M3->M4
LSHM(0): FRU 0 sensor 32 LUN 0 'HS 000 CM' hotswap M3->M4
........
pm_PwrFree(3)
bp_fru: power chan=1, current limit=8.0 A
PwrAllocate(fru 3 chan 1): allocating 1.0 A granted (available 37.5 A)
pm_EnablePayloadPwr(50,1):
PM(50): payload already ON for FRU 3
PM(50,1):Enable PP (primSite:1 secSite:0)
WARN - LSHM(0): ignore version change sensor
LSHM(0): CM sensor 59 LUN 0 <unknown> hotswap M1->M2
LSHM(0): FRU 3 sensor 66 LUN 0 'HotSwap' hotswap M2->M3
LSHM(0): FRU 3 sensor 66 LUN 0 'HotSwap' hotswap M3->M4
pm_processEvent PM(50): channel(1) new state=0x5b
LSHM(0): FRU 7 sensor 38 LUN 0 'HS 007 AMC3' hotswap M3->M4
LSHM(0): FRU 7 sensor 38 LUN 0 'HS 007 AMC3' hotswap M3->M4
ScanPM: ...
no PM for fru_id=51
no PM for fru_id=52
no PM for fru_id=53
LSHM(0): FRU 40 sensor 52 LUN 0 '+12V' voltage 'lower non-critical go low' -assertion 
LSHM(0): FRU 40 sensor 52 LUN 0 '+12V' voltage 'lower critical go low' -assertion 
LSHM(0): FRU 40 sensor 52 LUN 0 '+12V' voltage 'lower non-recoverable go low' -assertion 
LSHM(0): FRU 40 sensor 51 LUN 0 '+12V_1' voltage 'lower non-critical go low' -assertion 
LSHM(0): FRU 40 sensor 51 LUN 0 '+12V_1' voltage 'lower critical go low' -assertion 
LSHM(0): FRU 40 sensor 51 LUN 0 '+12V_1' voltage 'lower non-recoverable go low' -assertion 
```

But, whatever happens, Sayma never gets powered.

**Comment by sbourdeauducq** [comment]
This later error turns off the fans completely (and then parts of the super-reliable µTCA crate overheat), but the MCH still responds to TCP/IP.

**Comment by vdirksen** [comment]
@sbourdeauducq : Please start the web interface of the NAT-MCH. If you have not changed the IP address, then you can use "firefox 192.168.1.41". Login if not changed is "root" without quotes and password "nat" without quotes. Scroll down to "System information". Click on this and then let it create a text file. Please send that system file to support@nateurope.com. We can see then the history and also all components in the system. We will try to support you here.

**Comment by hartytp** [comment]
@gkasprow I gather this is fixed now, but please reopen if it's not.

**Comment by sbourdeauducq** [comment]
@hartytp No commits to https://github.com/m-labs/mmc-firmware so obviously this isn't fixed.

**Comment by hartytp** [comment]
Hmm Greg reported having BP Ethernet working on Sayma in a uTCA rack. So I think the issue is fixed in his code. Not having the code uploaded to the repo sounds like a separate issue to me.

**Comment by gkasprow** [comment]
@wizath where is the code?

**Comment by sbourdeauducq** [comment]
@gkasprow @wizath Did you get Sayma running without power supply or fan-related bugs in the µTCA crate? Did you test it very carefully, as the behavior I observed was non-deterministic?
If so, can we get updated OpenMMC firmware images?

**Comment by jbqubit** [comment]
I received two Sayma_AMC + Sayma_RTM from Warsaw this morning. When I put them into my uTCA crate and power it on the AMC lights flash synchronously green, red and blue. No power LEDs are illuminated on AMC PCB. @gkasprow @wizath Did this work in your crate prior to shipping to UMD? 

See attached for diagnostic info for my NAT Native R5 crate. 
[nat_mch_sysinfo.txt](https://github.com/sinara-hw/sinara/files/2090999/nat_mch_sysinfo.txt)
[nat_mch_running_cfg (3).txt](https://github.com/sinara-hw/sinara/files/2091000/nat_mch_running_cfg.3.txt)

**Comment by gkasprow** [comment]
Can you try with just one AMC board (without RTM) ?

**Comment by wizath** [comment]
> @gkasprow @wizath Did you get Sayma running without power supply or fan-related bugs in the µTCA crate? Did you test it very carefully, as the behavior I observed was non-deterministic?
> If so, can we get updated OpenMMC firmware images?

I've tested Sayma in two configurations:

- Vadatech crate & Vadatech MCH
- Vadatech crate & NAT MCH

On both MCH's board seems to work fine, it acquires power and hotswaps without any problems. 
I had fan issues, but these were related to Vadatech MCH incompatibility with their own FANs - they go crazy and full speed after about 2 mins of uptime.

Another issue is that sometimes RTM fails to get power rails up. This AFAIK is related to Sayma Duo current draw at initialization and because we're using only one PSU atm - although this is rare.

I've also encountered several events of whole uTCA crate dying unexpectedly. I think that is related to this strange transition between Bootloader mode and normal appliaction, since it occured while I was plugging the USB cable in - but that pattern is not really deterministic.

**Comment by marmeladapk** [comment]
@jbqubit Does it work not in crate? Perhaps something is wrong with the board itself.

**Comment by jbqubit** [comment]
In case of AMC alone in crate the front panel led flashing is different. Only green led flashes. 

I'll try on bench top next.

**Comment by jbqubit** [comment]
Power systems on both Sayma are fine when running on the bench top.

**Comment by jbqubit** [comment]
openMMC Starting!
build date Apr 19 2018 14:12:54
CPU clock = 80 MHz
Selected board: Sayma
Could not find FRU information in EEPROM, building a runtime info...
>AMC FRU Information:
	-Board info area:
		-Language Code: 0
		-Manuf time: 10463040
		-Manufacturer: Creotech
		-Name: Sayma AMC
		-Serial Number: 1
		-Part Number: SAYMA
		-File ID: AFCFRU
No Chassis info area
No internal use area
	-Product info area:
		-Language Code: 0
		-Manufacturer: Creotech
		-Name: Sayma AMC
		-Part Number: 00001
		-Version: v1.0
		-Asset Tag: No tag
		-Serial Number: 1
		-File ID: AFCFRU
	-Multirecord Area: 
		-Module Current: 8 A
		-Zone3 Compatibility code: 0x11223344
>AMC FRU total size: 220 bytes
RTM Board disconnected!
AMC Hotswap handle pressed!
openMMC Starting!
build date Apr 19 2018 14:12:54
CPU clock = 80 MHz
Selected board: Sayma
Could not find FRU information in EEPROM, building a runtime info...
>AMC FRU Information:
	-Board info area:
		-Language Code: 0
		-Manuf time: 10463040
		-Manufacturer: Creotech
		-Name: Sayma AMC
		-Serial Number: 1
		-Part Number: SAYMA
		-File ID: AFCFRU
No Chassis info area
No internal use area
	-Product info area:
		-Language Code: 0
		-Manufacturer: Creotech
		-Name: Sayma AMC
		-Part Number: 00001
		-Version: v1.0
		-Asset Tag: No tag
		-Serial Number: 1
		-File ID: AFCFRU
	-Multirecord Area: 
		-Module Current: 8 A
		-Zone3 Compatibility code: 0x11223344
>AMC FRU total size: 220 bytes
RTM Board disconnected!
AMC Hotswap handle pressed!




------------ PHY Dump ----------
Register, ADDR DATA
BMCR 0 0x0
BMSR 1 0x0
ID1 2 0x0
ID2 3 0x0
AN_ADV 4 0x0
AN_RX 5 0x0
AN_EXP 6 0x0
EXT_STAT 15 0x0
page0
JIT_DIAG 16 0x0
PCSCR 17 0x0
GMIICR 18 0x0
CR 19 0x0
IR 20 0x0
page1
ID 16 0x0
GPIOCR1 17 0x0
GPIOCR2 18 0x0
GPIOSR 19 0x0
PTPCR1 20 0x0
page2
PTPCR1 16 0x0
?? 17 0x0
?? 18 0x0
?? 19 0x0
?? 20 0x0
page3
?? 16 0x0
?? 17 0x0
?? 18 0x0
?? 19 0x0
?? 20 0x0
------------ Exar Dump ----------
GET_HOST_STS 0x2 0x0 0x0
GET_FAULT_STS 0x5 0x0 0x0
PWR_GET_STATUS 0x9 0x0 0x0
PWR_READ_VOLTAGE_CH1 0x10 0 0 V
PWR_READ_VOLTAGE_CH2 0x11 0 0 V
PWR_READ_VOLTAGE_CH3 0x12 0 0 V
PWR_READ_VOLTAGE_CH4 0x13 0 0 V
PWR_READ_VOLTAGE_IN 0x14 0 0 V

**Comment by gkasprow** [comment]
What is the MCH configuration? maybe the current limit is too low per AMC slot?

**Comment by jbqubit** [comment]
[nat_mch_running_cfg (4).txt](https://github.com/sinara-hw/sinara/files/2091647/nat_mch_running_cfg.4.txt)
[nat_mch_sysinfo.txt](https://github.com/sinara-hw/sinara/files/2091648/nat_mch_sysinfo.txt)

**Comment by jbqubit** [comment]
Also observe that the fans shutdown on the crate but the power remains applied to PSU and AMC. So they get hot. This seems very bad behavior. SB observed something similar on his Native R5. 

I got email from NAT inviting me to upgrade MCH firmware to 2.20.1. Should I do this?

**Comment by gkasprow** [comment]
Yes, do it please.

**Comment by gkasprow** [comment]
Can you try it in different slot? Place in the most left and then to the most right.
I just remembered...
Long time ago I had issues with inrush current that caused similar effects in MTCA.
That's why I added 20mR resistor at the power entry.
The effect was visible only in certain uTCA chassis and also depended on AMC position in the crate - the resistance between power module and the AMC slot differs from slot to slot. And it was visible only with NAT power supplies. We have Vadatech PSUs here, at the WUT, so it's quite possible that it did not manifest with them.
The solution could be simple - increase value of R367 to let's say 0.1Ohm.
Maybe the NAT PSU has some settings ? Vadatech one does.

**Comment by jbqubit** [comment]
I've tried both leftmost and rightmost slots. Just Sayma_AMC by itself. I've tried this is a pair of NAT chassis: Native R5 and R7. Still no power LEDs on Sayma AMC.

**Comment by jbqubit** [comment]
Didn't notice this before. When MCH gets to the stage of it's startup when power is applied to Sayma_AMC  (Sayma_AMC lights go on) the MCH front panel goes completely dark on R5 and on R7 it goes dark except for flashing blue led. Whatever Sayma_AMC is doing it's very much influencing MCH.

**Comment by jbqubit** [comment]
The MCH has a UART. I've configured it to dump a bunch of log info to the UART. The attached file shows what's logged as the MCH starts up with Sayma_AMC inserted but its handle extended. I then insert handle and MCH becomes unresponsive within about 1 second. One of the last messages emitted by the MCH is the following. I'll ping NAT to take a look at this post to the Issue. 

> pm_PP_good(6): Failure: power channel state=0x0b
pm_PP_good(6): Failure: power channel state=0x0b

[putty.log](https://github.com/sinara-hw/sinara/files/2091932/putty.log)

**Comment by gkasprow** [comment]
It looks like the Sayma consumes too much inrush current and it influences the primary power converter which supplies also MCH. 
MTCA specifies max inrush current. I had such case with AFCK in the past and it was solved by lowering of the power stage input capacitance which was close to 1mF. In case of Sayma I implemented additional series resistor to mitigate it. Tomorrow I will measure the inrush current.
The MTCA specification defines the max inrush current in following way:
![obraz](https://user-images.githubusercontent.com/4325054/41261619-d26b8382-6ddc-11e8-85c9-b60b096d4b06.png)

**Comment by gkasprow** [comment]
So in our case we should not exceed 40A of inrush current.  So maybe this 20mR resistor is too small and it should be roughly 0.2Ohm to make sure this current will never be exceeded.  But it will also cause losses of 3W at 4A which is not acceptable. The only solution I see is implementation of current limiting resistor and bypass P-FET that shorts it after 10 ms. But let's measure the inrush current firs.

**Comment by gkasprow** [comment]
@jbqubit can you print log from the power module? Is there such option?

**Comment by gkasprow** [comment]
OK, so I measured the input inrush current and it's roughly 75A:
![tek00124](https://user-images.githubusercontent.com/4325054/41281115-e41b7aca-6e30-11e8-8ee9-b5aca57d9d51.png)

**Comment by gkasprow** [comment]
After I changed the resistor to 0.2Ohm, the inrush current is 40A
![tek00126](https://user-images.githubusercontent.com/4325054/41281760-890d0db8-6e32-11e8-822f-ae0998a812fe.png)

**Comment by gkasprow** [comment]
I added 0.2R and parallel MOSFET
Now it looks much better:
Violet is  voltage on the PMOS gate. 
![tek00127](https://user-images.githubusercontent.com/4325054/41284212-f01d8478-6e38-11e8-8201-75fb38a8e3aa.png)

**Comment by gkasprow** [comment]
You can replace the series resistor with 0.2 Ohm and see if this helps. If
so you will add MOSFET and RC circuit that will limit losses on the
resistor.
12 cze 2018 03:24 "Joe Britton" <notifications@github.com> napisał(a):

**Comment by jbqubit** [comment]
Do you mean R314? 

R + MOSFET + RC seems fine. Or negative temperature coefficient (NTC) thermistor?

**Comment by gkasprow** [comment]
NTC will probably not work for such high currents. And they have some inertia so when you switch off and on the power, the current peak will persist. 
I installed IRF9317 because I had such in my lab. One can also use the same as already used in the project - Si4425DDY
But let's try with 0.2..0.3R first.
Connections:
MOSFET pins 1,2,3 - board power entry, AMC connector side
MOSFET pin 4 - gate, 10nF to pin 3, 10k to GND
MOSFET pins 5,6,7,7 - payload side, close to the SATA connector

**Comment by jbqubit** [comment]
Ok. I’ll try after lunch.

**Comment by vdirksen** [comment]
To get power module log, please enable "debug flag for power module". This will give extra output on MCH CLI. I will have a look to the system file provided earlier to see the firmware status of the power module.

**Comment by jbqubit** [comment]
I don't have a suitable MOSFET. Using 33 uH inductor I can slow inrush and get the AMC to power on and boot. However, when I add RTM the front panel lights on the AMC flash together in unison. 

To move forward with using Sayma in the crate I did the following.
- removed R314 from Sayma_AMC
- applied external power to each AMC using stand-alone J4 header 
- remove MCH from rack -- fan automatically comes on

**Comment by gkasprow** [comment]
33uH may be not enough.
It's probably the coil resistance that does the job. It also means that the inrush current seem to be the problem. The RTM power on switch has already inrush current protection. Can you make a movie that illustrates the problem?
I have to get the NAT PSU to my hands to diagnose the problem.

**Comment by gkasprow** [comment]
@jbqubit I  plugged your to our crate, plugged 3 Sayma boards and they  got powered without troubles. I tried one board with 0R020 series resistor, one with inductor that was attached and one with MOSFET soft-start. They all behave in identical way.
What is interesting, I cannot run more than 3 boards at a time. When I plug fourth and fifth, they got no power and corresponding LED is blinking on MCH. But when I pull the handle on module 1, 2 or 3, module 4 gets power. When I pull handle on another module, the fifth one gets power.
So it looks like the MCH does not have enough power to distribute. The PSU show little loading. 
@wizath please check what is the current level that Sayma is requesting. AFAIR it was 8A which would explain why the MCH does not assign power to more than 3 modules.
I've just realized that Joe sent us his MCH, will try it tomorrow.

**Comment by gkasprow** [comment]
I've ordered  NAT crate with RF backplane,  2x MCH and 4 PSU modules and various accessories. The procurement after 4 months finally published the call for tender. Hopefully I will get them within 8 weeks in my lab.

**Comment by wizath** [comment]
It is set to 8A per board

**Comment by gkasprow** [comment]
OK, so this explains the issue. But still, so far no idea why it does not work in USA.

**Comment by jbqubit** [comment]
> @jbqubit I plugged your PSU to our crate, plugged 3 Sayma boards and they got powered without troubles. 

That's good. But you're using same Vadatech power supply and MCH as before, right? 

>I cannot run more than 3 boards at a time. When I plug fourth and fifth, 
>they got no power and corresponding LED is blinking on MCH. 

On NAT MCH there's an option to set both maximum current per AMC and delayed power-up for each AMC. I set my MCH to wait a couple seconds before powering on each AMC to avoid inrush current overwhelming the PSU. Does Vadatech have a similar option? 

@wizath please check what is the current level that Sayma is requesting. AFAIR it was 8A which would explain why the MCH does not assign power to more than 3 modules.

My Sayma AMC+RTM consumed about 4A (12V) when fully powered up. 

> I've just realized that Joe sent us his MCH, will try it tomorrow.

Yes. Testing using NAT MCH and NAT PSU is what was missing before. Not sure if it matters but would be good to use transformer to deliver PSU 110 V so it's close to USA power conditions.

**Comment by gkasprow** [comment]
> 
> 
>     @jbqubit I plugged your PSU to our crate, plugged 3 Sayma boards and they got powered without troubles.
> 
> That's good. But you're using same Vadatech power supply and MCH as before, right?
> 
I'm using your PSU and NAT MCH. Only crate is from Vadatech but it is passive.

>     I cannot run more than 3 boards at a time. When I plug fourth and fifth,
>     they got no power and corresponding LED is blinking on MCH.
> 
> On NAT MCH there's an option to set both maximum current per AMC and delayed power-up for each AMC. I set my MCH to wait a couple seconds before powering on each AMC to avoid inrush current overwhelming the PSU. Does Vadatech have a similar option?

no idea, I'm using NAT one. I did not configure it  in this way

> 
> @wizath please check what is the current level that Sayma is requesting. AFAIR it was 8A which would explain why the MCH does not assign power to more than 3 modules.
> 
> My Sayma AMC+RTM consumed about 4A (12V) when fully powered up.
> 
>     I've just realized that Joe sent us his MCH, will try it tomorrow.
> 
> Yes. Testing using NAT MCH and NAT PSU is what was missing before. Not sure if it matters but would be good to use transformer to deliver PSU 110 V so it's close to USA power conditions. 

As I said, I used NAT MCH
I have a transformer so could check if no other ideas come to my mind.
What's surprising, the chassis fans that had troubles with Vadatech PSU, work fine with your PSU :D They are from Vadatech of course.

**Comment by gkasprow** [comment]
@jbqubit I plugged your MCH and your PSU and it also works fine.

**Comment by gkasprow** [comment]
[Here](https://www.dropbox.com/s/7jbzpk01psc1gen/2018-07-12%2010.59.56.mp4?dl=0) is some movie. @jbqubit was the MCH behaving in similar way in your place?

**Comment by jbqubit** [comment]
These were my observations. 
- https://github.com/sinara-hw/sinara/issues/475#issuecomment-396322137
- https://github.com/sinara-hw/sinara/issues/475#issuecomment-396339510
- https://github.com/sinara-hw/sinara/issues/475#issuecomment-396759392

There's still Vadatech crate and fan unit in your system. Is there at NAT salesperson in Warsaw who could loan you a NAT crate for a day or two? Or maybe that salesperson could tell you about another group at/near WUT that has a NAT crate you could borrow. 

Did you try running the PSU from 110 VAC?

**Comment by gkasprow** [comment]
@jbqubit We have NAT crate in Creotech, but I think I managed to recreate your problem.
In certain slot/AMC/RTM configuration I managed to get  situation that MCH dies after the AMC gets power on. Interestingly, this happens on AMC board with slow start mechanism implemented!
I'm happy that finally get stable not working configuration :)

**Comment by gkasprow** [comment]
@jbqubit I found what was the reason.
The slow-start circuit that was added to limit the inrush current of RTM was not enough. 100us rise time was too much for NAT PSU.
What's surprising, it is able to tollerate much higher inrush current but during the power on sequence.
RTM power on is negotiated with PSU independently and when MCH let the AMC turn on the RTM power, another inrush peak occurs causing the MCH to die. Luckily fans continue spinnig.

And it seems that we were on the edge of what PSU can accept because only certain combination of AMC and RTM triggered such problem.
And you were unlucky man who discovered that.
I increased the time constant to 2.2uF*1.5kOhm and now it's fine.

**Comment by jbqubit** [comment]
> I increased the time constant to 2.2uF*1.5kOhm and now it's fine.

Whew! Glad you were able to reproduce. 

If I understand correctly you've added a current limiting MOSFET at the power input on the AMC. The RTM in turn is powered in series, right? I'm worried about a voltage sag on the AMC when the RTM is cycled on. Is a second MOSFET (with slower time constant) at the 12V inlet to the RTM warranted? 

Please also check with 110 VAC as I suspect the PSU may have different current-sourcing properties.

**Comment by gkasprow** [comment]
There are 2 inrush current limit circuits - one is 0.2R resistor and MOSFET at the power entry of AMC, another is at the RTM rail, also on AMC. The first one works when MCH assigns power to AMC, the second works when RTM gets power from AMC. 
Without proper slow start circuit I observed multiple AMC supply restarts because the sag caused by starting the RTM was high enough to trigger current limit of my 5A bench supply.
After the fix I observe voltage sag is far smaller, roughly 1V and caused mainly by long cables to the bench supply.

**Comment by jbqubit** [comment]
OK. Glad to know that's covered. :)

**Comment by gkasprow** [comment]
@vdirksen we have interesting case with AC600D, NAT MCH and Vadatech MTCA crate.
After I modified Sayma AMCs to limit the inrush current it works fine in slots 1-9
In slots 10-12 when I press the handle, the MCH dies and so do all 12V rails.
Sometimes only one of the fan tray stops but rest of the cards work.
I looked at the inrush current of the AMC module and I removed the current limiting resistor. I also increased startup time to 1ms so the inrush current does not exceed 2A. And still the same, the MCH dies when AMC is in slots 10-12.
The same AMC board with disconnected payload power (removed inrush current limiter) does not kill the MCH, so it is not an issue with capacitors on the bakcplane.

**Comment by gkasprow** [comment]
When I move the PSU to tight most power slot, the situation is the same. So it does not seem to be an issue with trace resistance on the backplane

**Comment by gkasprow** [comment]
with 0.47Ohm series resistor it works fine in all slots. When I enable the MOSFET that shorts it after 4ms, the MCH fails...

**Comment by gkasprow** [comment]
I think I found what is going on. A few tens of ms after the MOSFET is enabled, I see large 100us-wide current peak.

**Comment by gkasprow** [comment]
but it is only 2A

**Comment by gkasprow** [comment]
It looks like the PM does not like such rapid current peaks. Vadatech PSU desn't care about them but NAT PSU is very sensitive.

**Comment by gkasprow** [comment]
And the problematic was P12V_FMC power switch. It caused such current peak. After I added RC circuit (easy to do) the peak is much lower, not so steep so it does not trigger the PSU failure.
So to summarize, there were 3 factors that caused uTCA operation failure.
- wrong capacitor in the RTM 12V switch inrush current limter. I discovered that in my compartment box (nice black ESD ones with lids) I had mixture of 1uF and 1nF caps. So during modification I did a year ago, some Sayma bards received 1uF, some 1nF... I thrown away all the caps.
- wrong value of current limiting resistor. It was 20mR, I'm not 100% sure if this was really an issue, but with that value we do not conform with uTCA specification. I changed it to 0R22 and added MOSFET in parallel that shorts it after a few ms.
- missing inrush current limiter for 12V supply. I modified 2 Sayma boards and they work reliably in all uTCA slots now. I will modify remaining boards and then can ship one to @sbourdeauducq

**Comment by gkasprow** [comment]
Here is a plot showing voltage drop on series 0.47Ohm resistor. The scope GND is connected to the +12V, so that's why voltage is negative. I had to use separation transformer to cope with PE mains line.
![tek00137](https://user-images.githubusercontent.com/4325054/42761691-9076f612-890e-11e8-88c6-280580ccd231.png)

**Comment by vdirksen** [comment]
@gkasprow : Just back from vacation I saw, that you identified the issues and solved it. Congratulation.

**Comment by gkasprow** [comment]
one of the boards I got from @sbourdeauducq failed to init in crate because 3.3V MP drawn too much current - over 0.25A from 3.3V. I figured it out that one of thermometers failed. No idea why I2C chip supplied from 3.3V got damaged. Could be ESD. After I replaced it with new one the board works.

**Comment by marmeladapk** [comment]
Another board we had was constantly resetting when plugged into crate (with one PM) with RTM (lights on FP blinking rapidly). After implementing slow start for 3V3 MP the issue dissapeared.

**Comment by hartytp** [comment]
@glasprow again, is there anything to do here before freezing v2.0 schematics?

**Comment by hartytp** [comment]
Ping @gkasprow

**Comment by jbqubit** [comment]
@gkasprow We'd like to know if there any further changes before freezing RTM schematics.

**Comment by gkasprow** [comment]
No, all issues with NAT MCH and supplies were caused by missing soft start circuits or excessive management power. We did not experience it before because we're using Vadatech stuff which does not care about inrush current.
